documentation/DESIGN.md

```diff
@@ -0,0 +1,249 @@
+# Design
+
+## Overview of the User Facing Classes
+
+Here is a diagram showing the relationship between the main collection types:
+
+![Hierarchy](http://www.plantuml.com/plantuml/proxy?src=https://raw.github.com/scala/collection-strawman/master/documentation/hierarchy.plantuml)
```

Comment:
This URL will only work when the PR is merged. Here is a woraround:

![](http://www.plantuml.com/plantuml/proxy?src=https://raw.github.com/scala/collection-strawman/0ede1d1f76ccd4fb24627fc39e47deecf681e73b/documentation/hierarchy.plantuml)